src/main/kotlin/app/cash/licensee/task.kt

```diff
@@ -18,56 +18,101 @@ package app.cash.licensee
 import kotlinx.serialization.builtins.ListSerializer
 import kotlinx.serialization.json.Json
 import org.gradle.api.DefaultTask
 import org.gradle.api.artifacts.Configuration
-import org.gradle.api.attributes.Usage
-import org.gradle.api.attributes.Usage.USAGE_ATTRIBUTE
-import org.gradle.api.file.FileCollection
+import org.gradle.api.artifacts.ResolvedArtifact
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.artifacts.result.ResolvedVariantResult
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.logging.LogLevel.ERROR
 import org.gradle.api.logging.LogLevel.INFO
 import org.gradle.api.logging.LogLevel.LIFECYCLE
 import org.gradle.api.logging.LogLevel.WARN
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
-import org.gradle.api.tasks.Classpath
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 
-internal open class LicenseeTask : DefaultTask() {
+internal abstract class LicenseeTask : DefaultTask() {
   @get:Input
-  lateinit var dependencyConfig: DependencyConfig
+  abstract val dependencyConfig: Property<DependencyConfig>
 
   @get:Input
-  lateinit var validationConfig: ValidationConfig
+  abstract val validationConfig: Property<ValidationConfig>
 
   @get:Input
-  lateinit var violationAction: ViolationAction
+  abstract val violationAction: Property<ViolationAction>
 
-  @get:Classpath
-  var classpath: FileCollection by Delegates.notNull()
-    private set
+  @get:Input
+  abstract val coordinatesToPomInfo: MapProperty<DependencyCoordinates, PomInfo>
+
+  fun addPomFileDependencies(configuration: Configuration) {
+    val root = configuration.incoming.resolutionResult.rootComponent
+
+    val pomInfos: Provider<Map<DependencyCoordinates, PomInfo>> = root.zip(dependencyConfig) { root, depConfig ->
+      val directDependencies = loadDependencyCoordinates(
+        logger,
+        root,
+        depConfig,
+      )
+      val directPomFiles = directDependencies.coordinates.fetchPomFiles(root.variants)
+      directPomFiles.getPomInfo(root.variants)
+    }
+
+    this.coordinatesToPomInfo.set(pomInfos)
+  }
 
-  @get:Internal
-  var configuration: Configuration by Delegates.notNull()
-    private set
+  private fun Iterable<ResolvedArtifact>.getPomInfo(variants: List<ResolvedVariantResult>): Map<DependencyCoordinates, PomInfo> {
+    val factory = DocumentBuilderFactory.newInstance()
+    val documentBuilder = factory.newDocumentBuilder()
 
-  fun setClasspath(configuration: Configuration, usage: String) {
-    this.configuration = configuration
+    return associate { pom ->
+      // cast is safe because all resolved artifacts are pom files
```

Comment:
```suggestion
      // Cast is safe because all resolved artifacts are pom files.
```

We follow https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html